### PR TITLE
Add pkg:lints to native_app, FFI fixes

### PR DESCRIPTION
**Pull request by domesticmouse**

### CODE REVIEW

**Comment by domesticmouse** [comment]
@johnpryan for context, I dropped the `Dockerfile` recipes from the FFI examples, as Docker can't deal with file imports from outside the Docker source directory. The FFI projects have the following dependecy which breaks under Docker:

```yaml
dev_dependencies:
  test_utils:
    path: ../test_utils
```

**Comment by mit-mit** [comment]
Isn't this a dupe of https://github.com/dart-lang/samples/pull/112 ?

**Comment by domesticmouse** [comment]
I didn't see #112, but after updating there are still changes outstanding for FFI. I'll rename this PR

File: ffi/system-command/win32ui.dart [hunk 1]
```diff
@@ -31,6 +31,7 @@ typedef MessageBoxDart = int Function(
   int type,
 );
 
+// ignore_for_file: constant_identifier_names
```
Comment (mit-mit) [inline ffi/system-command/win32ui.dart:34]: Maybe add this to the `analysis_options.yaml` instead?

**Review by mit-mit — APPROVED** [review]

File: ffi/system-command/win32ui.dart (hunk 1 above)
Comment (domesticmouse) [reply ffi/system-command/win32ui.dart]: It looks like John added it, so I dropped the redundant comments.

**Comment by domesticmouse** [comment]
@johnpryan and @mit-mit, any idea what is required to get the **analyze; linux; Dart stable; PKGS: extension_methods, ffi/primitives, ffi/hello_world,...** required test to pass? It's blocking landing this PR.

**Comment by mit-mit** [comment]
I don't know where that checks is coming from. It's really weird that there is no "Details" link for it.

File: ffi/system-command/win32ui.dart (hunk 1 above)
Comment (johnpryan) [reply ffi/system-command/win32ui.dart]: SGTM

**Comment by johnpryan** [comment]
That run isn't showing up on the details page https://github.com/dart-lang/samples/pull/115/checks?check_run_id=2727468020

**Review by johnpryan — APPROVED** [review]

**Comment by domesticmouse** [comment]
Hey @RedBrogdon, can you give some insight here?

**Pull request merged**